src/app/(authenticated)/(dashboard)/[publicId]/settings/profile/page.tsx

```diff
@@ -4,81 +4,237 @@ import Image from "next/image";
+  loginEmail: z.string().email(),
+  workEmail: z.string().email(),
+  avatarUrl: z.string(),
+});
 
 const ProfileSettingsPage = () => {
-  const { data } = useSession();
-  const form = useForm();
+  const { toast } = useToast();
+  const utils = api.useUtils();
+  const memberProfile = api.member.getProfile.useQuery();
 
+  const [file, setFile] = useState<File | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  // @ts-expect-error: xxxx
+  const form = useForm<z.infer<typeof profileSettingsSchema>>({
+    resolver: zodResolver(profileSettingsSchema),
+  });
+
+  // @ts-expect-error: xxxx
+  const { errors } = form.formState;
+  const isSubmitting = form.formState.isSubmitting;
+  const isDirty = form.formState.isDirty;
+  const canSave: boolean = file ? isDirty || !!file : isDirty;
+
+  // @ts-expect-error: xxxx
+  const saveProfileMutation = api.member.updateProfile.useMutation({
+    onSuccess: async () => {
+      await utils.member.getProfile.invalidate();
+      return toast({
+        variant: "default",
+        title: "Profile changed successfully.",
+      });
+    },
+    onError: () => {
+      return toast({
+        variant: "destructive",
+        title: "Failed updating profile",
+        description: "Something went wrong.",
+      });
+    },
+  });
+
+  useEffect(() => {
+    if (memberProfile?.data) {
+      // @ts-expect-error: xxxx
+      form.reset(memberProfile?.data);
+    }
+  }, [memberProfile?.data, form]);
+
+  const handleImageUpload = async (
+    _file: File,
+  ): Promise<{ imageUrl: string }> => {
+    // const options = {
+    //   expiresIn: 3600,
+    //   keyPrefix: "opencap",
+    //   identifier: "unique-id",
+    // };
+    // const { key, ...rest } = await uploadFile(file, options);
+    // const imageUrl = `${process.env.UPLOAD_ENDPOINT}/${key}`;
+    return { imageUrl: "https://avatars.githubusercontent.com/u/456802?v=4" };
+  };
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files;
+
+    if (files?.[0]) {
+      const maxSizeInBytes = 1024 * 1024; // 1 MB
+
+      const allowedTypes = ["image/jpeg", "image/png", "image/gif"];
+
+      if (files[0].size > maxSizeInBytes) {
+        return toast({
+          variant: "destructive",
+          title: "File limit exceeded.",
+          description: "File size exceeds the maximum allowed(1 MB)",
+        });
+      }
+      if (!allowedTypes.includes(files[0].type)) {
+        return toast({
+          variant: "destructive",
+          title: "Invalid file format.",
+          description: "Allowed types: JPEG, PNG, GIF",
+        });
+      }
+      setFile(files[0]);
+    }
+  };
+  const handleAvatarChange = () => {
+    if (fileInputRef.current) {
+      fileInputRef.current.click();
+    }
+  };
+  const handleRemoveAvatar = () => {
+    setFile(null);
+  };
+  async function onSubmit(values: z.infer<typeof profileSettingsSchema>) {
```

Comment:
Changing a profile picture and saving profile information are two different actions like I said the day before. Clicking on save button should only save the profile information.